sphinx/ext/autodoc/__init__.py

```diff
@@ -1485,13 +1494,32 @@ def add_directive_header(self, sig: str) -> None:
         if not self.doc_as_attr and self.options.show_inheritance:
             sourcename = self.get_sourcename()
             self.add_line('', sourcename)
-            if hasattr(self.object, '__bases__') and len(self.object.__bases__):
-                bases = [':class:`%s`' % b.__name__
-                         if b.__module__ in ('__builtin__', 'builtins')
-                         else ':class:`%s.%s`' % (b.__module__, b.__qualname__)
-                         for b in self.object.__bases__]
-                self.add_line('   ' + _('Bases: %s') % ', '.join(bases),
-                              sourcename)
+
+            def restify(cls: "Type") -> str:
```

Comment:
This helper fails when a generic base class takes a `Union` object as a parameter.